pyomo/core/base/sos.py

```diff
@@ -106,48 +106,180 @@ def set_items(self, variables, weights):
 @ModelComponentFactory.register("SOS constraint expressions.")
 class SOSConstraint(ActiveIndexedComponent):
     """
-    Represents an SOS-n constraint.
-
-    Usage:
-    model.C1 = SOSConstraint(
-                             [...],
-                             var=VAR,
-                             [set=SET OR index=SET],
-                             [sos=N OR level=N]
-                             [weights=WEIGHTS]
-                             )
-        [...]   Any number of sets used to index SET
-        VAR     The set of variables making up the SOS. Indexed by SET.
-        SET     The set used to index VAR. SET is optionally indexed by
-                the [...] sets. If SET is not specified, VAR is indexed
-                over the set(s) it was defined with.
-        N       This constraint is an SOS-N constraint. Defaults to 1.
-        WEIGHTS A Param representing the variables weights in the SOS sets.
-                A simple counter is used to generate weights when this keyword
-                is not used.
-
-    Example:
-
-      model = AbstractModel()
-      model.A = Set()
-      model.B = Set(A)
-      model.X = Var(B)
-
-      model.C1 = SOSConstraint(model.A, var=model.X, set=model.B, sos=1)
-
-    This constraint actually creates one SOS-1 constraint for each
-    element of model.A (e.g., if |A| == N, there are N constraints).
-    In each constraint, model.X is indexed by the elements of
-    model.B[a], where 'a' is the current index of model.A.
-
-      model = AbstractModel()
-      model.A = Set()
-      model.X = Var(model.A)
-
-      model.C2 = SOSConstraint(var=model.X, sos=2)
-
-    This produces exactly one SOS-2 constraint using all the variables
-    in model.X.
+    Implements constraints for special ordered sets (SOS).
+
+    Parameters
+    ----------
+    sos : int
+        The type of SOS.
+    var : pyo.Var
+        The set of variables from which the SOS(s) will be created.
+    index : pyo.Set, list (non-indexed SOS) or dict (indexed SOS), optional
+        A data structure with the indexes for the variables that are to be
+        members of the SOS(s). The indexes can be provided as a pyomo Set:
+        either indexed, if the SOS is indexed; or non-indexed, otherwise.
+        Alternatively, the indexes can be provided as a list, for a non-indexed
+        SOS, or as a dict, for indexed SOS(s).
+    weights : pyo.Param or dict, optional
+        A data structure with the weights for each member of the SOS(s). These
+        can be provided as pyo.Param or as a dict. If not provided, these will
+        be determined automatically using their position in the structure.
```

Comment:
This is a bit vague.  I would suggest "If not provided, variable weights will be determined by the `var` index set"